docs/adapters/index.md

```diff
@@ -43,6 +43,136 @@ However, sometimes you need to access a feature specific to one of the adapters
 When that happens, you can pass a block when specifying the adapter to customize it.
 The block parameter will change based on the adapter you're using. See each adapter page for more details.
 
+## Write your own adapter
+
+Adapters have methods that can help you implement support for a new backend.
+
+This example will use a fictional HTTP backend gem called `FlorpHttp`. It doesn't
+exist. Its only function is to make this example more concrete.
+
+### An Adapter _is_ a Middleware
+
+There are only two things which are actually mandatory for an adapter middleware to function:
+
+- a `#call` implementation
+- a call to `#save_response` inside `#call`, which will keep the Response around.
+
+These are the only two things.
+
+The rest of this text is about methods which make the authoring easier.
+
+### Helpful class: `::Faraday::Adapter`
```

Comment:
Subclassing `Faraday::Adapter` is also necessary to access `save_response` and to be recognised as an adapter by the rack builder, so I'd make that a requirement.